Document store tests raise error if order of returned labels not as expected

**Describe the bug**
During label tests in `test_document_store.py` there are several calls that request `labels` from the document store that have been built from two sets of labels (`label` and `label2`). Some of these calls are followed by `delete_labels` calls that specify the ID of a label to delete via the returned `labels` variable, eg they use:

```python
# delete filtered label2 by id
document_store.delete_labels(ids=[labels[1].id])
```

Rather than:

```python
# delete filtered label2 by id
document_store.delete_labels(ids=[label2.id])
```

Maybe there is a reason for this, but I found when testing this on the Pinecone doc store (which does not seem to return labels in the expected order), it causes the test to fail as the above call would actually delete the `label` entry rather than the `label2` entry.

**Error message**
```python
AssertionError: label != labels[0]
```

**Expected behavior**
The label to be deleted should be specified based on `label` and `label2` variables.

**Additional context**
...

**To Reproduce**
Run tests on Pinecone doc store from #2749 

**FAQ Check**
- [x] Have you had a look at [our new FAQ page](https://haystack.deepset.ai/overview/faq)?

**System:**
 - OS: MacOS M1
 - GPU/CPU: CPU
 - Haystack version (commit or version number): #2749 
 - DocumentStore: Pinecone
 - Reader: N/A
 - Retriever: N/A

--- a/test/document_stores/test_document_store.py
+++ b/test/document_stores/test_document_store.py
@@ -793,7 +793,7 @@ def test_labels(document_store: BaseDocumentStore):
     assert label2 in labels
 
     # delete filtered label2 by id
-    document_store.delete_labels(ids=[labels[1].id])
+    document_store.delete_labels(ids=[label2.id])
     labels = document_store.get_all_labels()
     assert label == labels[0]
     assert len(labels) == 1
@@ -804,7 +804,7 @@ def test_labels(document_store: BaseDocumentStore):
     assert len(labels) == 2
 
     # delete filtered label2 by query text
-    document_store.delete_labels(filters={"query": [labels[1].query]})
+    document_store.delete_labels(filters={"query": [label2.query]})
     labels = document_store.get_all_labels()
     assert label == labels[0]
     assert len(labels) == 1
@@ -815,7 +815,7 @@ def test_labels(document_store: BaseDocumentStore):
     assert len(labels) == 2
 
     # delete intersection of filters and ids, which is empty
-    document_store.delete_labels(ids=[labels[0].id], filters={"query": [labels[1].query]})
+    document_store.delete_labels(ids=[label.id], filters={"query": [label2.query]})
     labels = document_store.get_all_labels()
     assert len(labels) == 2
     assert label in labels

